tests/unit/test_type_match.py

```diff
@@ -30,6 +31,17 @@ class SubC(C): ...
 class D: ...
 
 
+class MyEnum(enum.StrEnum):
+    A = "a"
+
+
+T4 = TypeVar("T4", bound=MyEnum)
+
+
+class B[T: MyEnum](Protocol):
```

Comment:
Do we really need Enum here to encounter an issue? Does it work with any constraint like `T: int`?